test/powershell/engine/Basic/ValidateAttributes.Tests.ps1

```diff
@@ -260,4 +260,124 @@ Describe 'Validate Attributes Tests' -Tags 'CI' {
                 $ScriptBlock | Should Not Throw
         }
     }
+
+    Context "ValidateNotNull, ValidateNotNullOrEmpty and Not-Null-Or-Empty check for Mandatory parameter" {
+
+        BeforeAll {
+            function MandType {
+                param(
+                    [Parameter(Mandatory, ParameterSetName = "ByteArray")]
+                    [byte[]] $ByteArray,
+
+                    [Parameter(Mandatory, ParameterSetName = "ByteList")]
+                    [System.Collections.Generic.List[byte]] $ByteList,
+
+                    [Parameter(Mandatory, ParameterSetName = "ByteCollection")]
+                    [System.Collections.ObjectModel.Collection[byte]] $ByteCollection,
+
+                    [Parameter(ParameterSetName = "Default")]
+                    $Value
+                )
+            }
+
+            function NotNullFunc {
+                param(
+                    [ValidateNotNull()]
+                    $Value,
+                    [string] $TestType
+                )
+
+                switch ($TestType) {
+                    "COM-Enumerable" { $Value | ForEach-Object Name }
+                    "Enumerator"     {
+                        $items = foreach ($i in $Value) { $i }
+                        $items -join ","
+                    }
+                }
+            }
+
+            function NotNullOrEmptyFunc {
+                param(
+                    [ValidateNotNullOrEmpty()]
+                    $Value,
+                    [string] $TestType
+                )
+
+                switch ($TestType) {
+                    "COM-Enumerable" { $Value | ForEach-Object Name }
+                    "Enumerator"     {
+                        $items = foreach ($i in $Value) { $i }
+                        $items -join ","
+                    }
+                }
+            }
+
+            $filePath  = Join-Path -Path $PSHOME -ChildPath System.Management.Automation.dll
+            $byteArray = [System.IO.File]::ReadAllBytes($filePath)
+            $byteList  = [System.Collections.Generic.List[byte]] $byteArray
+            $byteCollection = [System.Collections.ObjectModel.Collection[byte]] $byteArray
+            ## Use the running time of 'MandType -Value $byteArray' as the baseline time
+            $baseline = (Measure-Command { MandType -Value $byteArray }).Milliseconds
+            ## Running time should be less than 'expected'
+            $expected = $baseline + 20
+
+            if ($IsWindows) {
+                $null = New-Item -Path $TESTDRIVE/file1 -ItemType File
```

Comment:
-ItemType File is not necessary anymore. The default itemtype is file

Reply:
OK. I will remove it.